packages/space-opera/src/components/animation_controls/reducer.ts

```diff
@@ -15,34 +15,22 @@
  *
  */
 
-import {registerStateMutator, State} from '../../space_opera_base.js';
+import {Action, State} from '../../types.js';
 
-/** Set auto play enabled or not */
-const SET_AUTOPLAY_ENABLED = 'SET_AUTOPLAY_ENABLED';
-export const dispatchAutoplayEnabled = registerStateMutator(
-    SET_AUTOPLAY_ENABLED, (state: State, enabled?: boolean) => {
-      state.config = {...state.config, autoplay: !!enabled};
-    });
+const SET_ANIMATION_NAMES = 'SET_ANIMATION_NAMES';
+export function dispatchSetAnimationNames(animationNames: string[]) {
+  return {type: SET_ANIMATION_NAMES, payload: animationNames};
+}
 
-/** Set animation name */
-const SET_ANIMATION_NAME = 'SET_ANIMATION_NAME';
-export const dispatchAnimationName = registerStateMutator(
-    SET_ANIMATION_NAME, (state: State, animationName?: string) => {
-      // Allow animationName === undefined to unset animationName
-      if (animationName && state.animationNames.indexOf(animationName) === -1) {
-        return;
-      }
+export const getAnimationNames = (state: State) =>
+    state.entities.modelViewerSnippet.animationNames;
 
-      state.config = {
-        ...state.config,
-        animationName,
-      };
-    });
-
-/** Set playAnimation or not */
-const PLAY_ANIMATION = 'PLAY_ANIMATION';
-export const dispatchPlayAnimation = registerStateMutator(
-    PLAY_ANIMATION, (state: State, playAnimation?: boolean) => {
-      // No need to copy state - we're always given a new copy.
-      state.playAnimation = !!playAnimation;
-    });
\ No newline at end of file
+export function animationNamesReducer(
```

Comment:
this "boilerplate" you'll see throughout, I think adds to readability and is common among redux applications

Reply:
default is how it is initialized, (and maintained when the action isn't associated with this state) 

the way redux works, is that all of the these reducers are called for each state update, and most are just ignored